tensorbay/client/gas.py

```diff
@@ -448,7 +448,15 @@ def upload_dataset(
 
             dataset_client.update_notes(**dataset.notes)  # type: ignore[arg-type]
 
-            with Tqdm(sum(len(segment) for segment in dataset), disable=quiet) as pbar:
+            if isinstance(dataset, Dataset):
+                data_length = sum(len(segment) for segment in dataset)
```

Comment:
`data_length` should mean the length of data
just like `segment_length` means the length of segment
what you mean is the number of data in the dataset. you should choose another precise variable name